### ISSUE TITLE
Self-signed cert on server

### ISSUE THREAD

**Issue by jaslo** [alexrsagen/node-es6-request#9]
I was hoping I could pass an object in place of the url string, like in "Request". And for a self-signed cert, I could do this:
    return request.get({
        host: url.hostname,
        port: 443,
        path: url.pathname,
        method: 'GET',
        rejectUnauthorized: false,
        requestCert: true,
        agent: false
    })

I added an options parameter with just the cert-related fields and it almost worked -- except in res.on("end"), you call "this.destroy" and Request doesn't have a destroy method.
Maybe you meant super.destroy? or this._destroy?

**Comment by jaslo** [comment]
Looks like Readable.destroy was added in node 8 and I am running node 7 (which should be es6)!

**Comment by alexrsagen** [comment]
Sorry about the late response!

This library does not support node 7, as stated in package.json `engines.node` value. If you look at prior issues, i do not mean `super.destroy` (as that would be a static method) nor `this._destroy`, as `stream.Duplex` implements a method `destroy` that in turn calls `_destroy` which is intended to be defined by derivative classes. However, the aforementioned `destroy` method is **not** available in node versions before 8.0.0.

To get around the issue of self-signed certs on a server, first of all i'd recommend [Let's Encrypt](https://letsencrypt.org/). But as i see that may be impossible in certain situations, here is how you would pass `rejectUnauthorized` using this library:

```javascript
const request = require("es6-request");

request.get("https://self-signed.badssl.com/", {
    rejectUnauthorized: false
})
.then(([body, res]) => {
    console.log(body);
    // should work!
})
.catch(e => console.error(e)); // should not be called unless there is some other error
```
